### ISSUE TITLE
Arch Linux support for package inventory and vulnerability detection

### ISSUE THREAD

**Issue by avielw** [wazuh/wazuh#7075]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0.3-4014 | syscollector and vuln-detector | Manager and Agent | Packages and Sources | Arch Linux |

Hey Guys,
As I wrote on your stack community page, and as you know, wazuh does not support Arch Linux:
- The syscollector module does not support the pacman package manager
- The vuln-detector module doesn't have a vulnerability feed for Arch Linux.

I have written those missing features.
#7058 

### Update

We will make this an EPIC issue since we are working on the remaining tasks to make this improvement compatible with the Wazuh current code. These main tasks are:

- Refactor the Syscollector changes for the new data provider module
- Vulnerability Detector minor changes
- Add unit tests to cover all the additions
- Documentation

In addition, these other parallels developments can generate conflicts with these changes so we have to keep an eye on them:

- **External dependencies are now precompiled**: https://github.com/wazuh/wazuh/issues/7586
- **Vulnerebility Detector architecture is being updated**: https://github.com/wazuh/wazuh/issues/7479 and https://github.com/wazuh/wazuh/issues/7749

#7058 has been merged into [7075-arch-linux-support](https://github.com/wazuh/wazuh/tree/7075-arch-linux-support) to continue the development of the tasks described above.

**Comment by pereyra-m** [comment]
Hello @avielw !
Thank you very much for this fantastic contribution! We really appreciate your work.

I have been reading the interesting discussion and analysis in Slack about this with @chemamartinez [(thread)](https://wazuh.slack.com/archives/C0A933R8E/p1609360412123000). I'll reproduce the main parts below because it can get lost there.

I have linked this issue to the PR you have created in this repository.
And I would like to mention the work you are doing in the packages repository too:

- **Issue:**  Arch Linux support [#593](https://github.com/wazuh/wazuh-packages/issues/593)
- **PR:** Adding some support for Arch Linux agent [594](https://github.com/wazuh/wazuh-packages/pull/594)

Finally, I have briefly tested your branch with a CentOS 7 Manager/Arch Agent.
It was able to find the correct versions of the packages and report the vulnerabilities. If you have false positives, could you give and example?

Best regards.

#### Conversation summary
**Chema Martinez**

- It is better not to use a compilation flag like USE_ALPM for detecting when the agent is compiled for Arch Linux. The code to get Arch Linux packages should be compiled always, and the agent should detect where it is located in runtime, by reading the file /etc/arch-release

- It is amazing that Arch Linux provides a JSON feed with their vulnerabilities status. However, the vulnerability scan against that feed should be complemented with the one against the NVD, as for the rest of Linux platforms. I see you skip that stage [here](https://github.com/wazuh/wazuh/pull/7058/files#diff-5be796c9d0ece797269dcb96948ef6ba41e42211ebea03355e9ac2c1356b547aR2046).

**Aviel Warschawski**

- The reason I've added the USE_ALPM flag is because the code I've added requires the libalpm package and will not compile without it. So what do you think I should do? add libalpm as an external package like openssl, libyaml etc?

- Also, the reason I removed the NVD feed is because I got a lot of false positives. I think it happens because the versions are not always exactly the same in both feeds, but I'll try to understand better what exactly is happening.

**Chema Martinez**

- Taking into account the similar cases in the past, we prefer to include the libalpm (hoping it won’t be too heavy) instead of forcing that the Arch Linux agent should be compiled in an Arch Linux platform, since the packages for linux are unified

- The Vulnerability Detector engine should be prepared to filter the false positives by correlating results from both scans (Arch Linux feed vs NVD). We can help you with this part of debugging why it produces false positives.

**Comment by avielw** [comment]
I think it does work without any false positives right now, but I don't use the NVD feed.
The problem is that Chema said that we should use the NVD feed for Arch too, and if I do use it, I get a lot of false positives (I also sent a log with a lot of false positives to Chema earlier today).

**Comment by chemamartinez** [comment]
Hi guys,

The external packages are added to the official repository once the pull request is approved and ready to be merged. Meanwhile, they can be pulled from a temporary repository.

**Comment by avielw** [comment]
Okay, if that is the case I think everything works as it should. I'm removing the "draft" flag from the pull requests.

**Comment by chemamartinez** [comment]
Great @avielw!

In that case, we will review your pull request ASAP.

Best regards.

**Comment by Dwordcito** [comment]
Hi @avielw , the part of syscollector module, it should be refactored, since the complete implementation was changed, today it is C++.

Therefore, this part should be modified.

**Comment by avielw** [comment]
Hi @Dwordcito,
Well it makes sense that the syscollector module has changed, since 1282 commits has been added to the master repository since I've changed the syscollector, and my pull request is just been reviewed now.

**Comment by chemamartinez** [comment]
Hi there,

We will make this an EPIC issue since we are working on the remaining tasks to make this improvement compatible with the Wazuh current code. These main tasks are:

- Refactor the Syscollector changes for the new data provider module
- Vulnerability Detector minor changes
- Add unit tests to cover all the additions
- Documentation

Best regards.

**Comment by chemamartinez** [comment]
This development is finished by #8178.

Thank you again @avielw for your contribution!

Regards.
